migrations/20160524_fix_challenges.js

```diff
@@ -0,0 +1,237 @@
+'use strict';
+
+const authorName = 'Blade';
+const authorUuid = '75f270e8-c5db-4722-a5e6-a83f1b23f76b';
+
+global.Promise = require('bluebird');
+const MongoClient = require('mongodb').MongoClient;
+const chalk = require('chalk');
+
+const logger = {
+  info: _logger('info', 'cyan'),
+  success: _logger('info', 'green'),
+  error: _logger('error', 'red'),
+  log: _logger('log', 'white'),
+  warn: _logger('warn', 'yellow'),
+}
+
+function _logger (type, color) {
+  return function () {
+    let args = Array.from(arguments).map(arg => chalk[color](arg));
+    console[type].apply(null, args);
+  }
+}
+
+// PROD: Enable prod db
+// const NODE_DB_URI = 'mongodb://username:password@dsXXXXXX-a0.mlab.com:XXXXX,dsXXXXXX-a1.mlab.com:XXXXX/habitica?replicaSet=rs-dsXXXXXX';
+const NODE_DB_URI = 'mongodb://localhost/new-prod-copy';
+
+// Cached ids from running the findBrokenChallengeTasks query on a local copy of the db
+const TASK_IDS = require('../taskids.json');
+
+let db;
+let count = 0;
+
+var timer = setInterval(function(){
+  count++;
+  if (count % 30 === 0) {
+    logger.warn('Process has been running for', count / 60, 'minutes');
+  }
+}, 1000);
+
+connectToDb()
+  .then(findBrokenChallengeTasks)
+  .then(getDataFromTasks)
+  .then(getUserChallenges)
+  .then(getChallengeTasks)
+  .then(correctUserTasks)
+  .then(updateTasks)
+  .then(closeDb)
+  .catch(reportError)
+
+function connectToDb () {
+  return new Promise((resolve, reject) => {
+    MongoClient.connect(NODE_DB_URI, (err, database) => {
+      if (err) {
+        logger.error('Uh oh... Problem connecting to the new database');
+        return reject(err);
+      }
+
+      logger.success('Connected to the database');
+
+      db = database;
+
+      resolve(db);
+    });
+  });
+}
+
+function reportError (err) {
+  logger.error('Uh oh, an error occurred');
+  closeDb();
+  throw err;
+}
+
+function unique (array) {
+  return Array.from(new Set(array));
+}
+
+function findBrokenChallengeTasks () {
+  logger.info('Looking for broken tasks...');
+
+  // return db.collection('tasks').find({'challenge.broken': 'CHALLENGE_TASK_NOT_FOUND'}).toArray()
```

Comment:
If you don't want to use the pre-cached ids, uncomment this and comment the line below.